src/components/NavigationBar.js

```diff
@@ -1,39 +1,59 @@
-                <Box sx={{ cursor: 'pointer' }} onClick={() => navigate('/')}>
-                    <img src={logo} alt="logo" width="183px"/>
-                </Box>
-                <Stack direction="row" spacing={1} alignItems="center">
-                    <IconButton aria-label="home" size="large" onClick={() => navigate('/')}>
-                        <HomeOutlinedIcon sx={{ fontSize: '43px', color: 'black' }} />
-                    </IconButton>
-                    <IconButton aria-label="create post" size="large" onClick={() => console.log('Open modal for post adding!')}>
-                        <AddBoxOutlinedIcon sx={{ fontSize: '43px', color: 'black' }} />
-                    </IconButton>
-                    <IconButton aria-label="profile" size="large" onClick={() => navigate(`/profiles/${userId}`)}>
-                        <AccountCircleIcon sx={{ fontSize: '43px', color: 'black' }} />
-                    </IconButton>
-                </Stack>
-            </Stack>
+  return (
+    <Box height="80px" display="flex" paddingX={3} alignItems="center" borderBottom="1px solid #dadada" bgcolor="white">
+      <Stack direction="row" justifyContent="space-between" width="100%" marginX="auto">
+        <Box sx={{ cursor: 'pointer' }} onClick={() => navigate('/')}>
+          <img src={logo} alt="logo" width="183px" />
+        </Box>
+        <Stack direction="row" spacing={1} alignItems="center">
+          <IconButton aria-label="home" size="large" onClick={() => navigate('/')}>
+            <HomeOutlinedIcon sx={{ fontSize: '43px', color: 'black' }} />
+          </IconButton>
+          <IconButton aria-label="create post" size="large" onClick={() => setOpen(true)}>
+            <AddBoxOutlinedIcon sx={{ fontSize: '43px', color: 'black' }} />
+          </IconButton>
+          <IconButton aria-label="profile" size="large" onClick={() => navigate(`/profiles/${userId}`)}>
+            <AccountCircleIcon sx={{ fontSize: '43px', color: 'black' }} />
+          </IconButton>
+        </Stack>
+      </Stack>
+      <Modal
+        open={open}
+        onClose={() =>setOpen(false)}
+        aria-labelledby="modal-modal-title"
+        aria-describedby="modal-modal-description"
+        sx={{margin: 'auto'}}
+      >
+        <Box style={{height: "400px", width: '400px', backgroundColor: 'white', margin: '8% 35%', borderRadius: '10px', border: '1px outset black'}}>
+          <Stack direction='row' justifyContent='space-between' borderBottom='2px solid black'>
+            <p/>
+            <p>Create new post</p>
+            <IconButton aria-label="home" size="large" onClick={() => setOpen(false)}>
+            <CloseIcon sx={{ fontSize: '30px', color: 'black', justifyContent: 'flex-end' }} />
+          </IconButton>
+          </Stack>
+          <input type="file" placeholder="Create new post" onChange={(cos) => console.log(cos)} style={{height: "400px", width: '400px'}}/>
```

Comment:
Please don't hardcode input height (it wont be responsive in any way and it exceeds parent bounds), please use flex to make it take as much space as available